### ISSUE TITLE
Please provide documentation on how to get the Trace button in the portal GUI to work

### ISSUE THREAD

**Issue by qfyra** [Azure/api-management-self-hosted-gateway#297]
### Proposal

Tracing works using a managed gateway. When using a self-hosted gateway, it does not seem tracing works out of the box, as regardless of config, the message "Trace is not available because response does not contain Apim-Trace-Id header" is shown when pressing the button. Please provide documentation on how to get the Trace button in the portal GUI to work.

### Use-Case

Already works with managed gateway. Should work with self-hosted gateway. And although crafting API calls with e.g. REST Client and setting Ocp-Apim-Trace: true is possible, it is vastly more cumbersome compared to having the corresponding functionality readily available in the Trace button.

**Comment by tomkerkhove** [comment]
The trace button only works for managed gateway given the portal does not know where your self-hosted gateway is running.

New tracing API is not supported yet in self-hosted gateway.

Does that answer your question?

**Comment by qfyra** [comment]
Yes it does, thank you. I was under the impression that tracing via the portal GUI was supposed to work with self-hosted gateways, but I'm not sure why I was under that impression 😄 

Is there a Github issue that I can add a notification on for the new tracing API?

**Comment by tomkerkhove** [comment]
Not for now but we include things in the release notes

**Comment by J0F3** [comment]
@tomkerkhove 

> New tracing API is not supported yet in self-hosted gateway.

What is the idea to do tracing on a self-hosted gateway then? Is it not possible at all or do we need to use the old Method with the  `Ocp-Apim-Trace` Header and Subscription Key (which is not even documented anymore respectively the documentation states that it is not supported anymore: [API Management no longer supports subscriptions for tracing or the Ocp-Apim-Trace header.](https://learn.microsoft.com/en-us/azure/api-management/api-management-howto-api-inspector#prerequisites  )?

**Comment by tomkerkhove** [comment]
The old tracing API still works so you should not encounter any issues. New Tracing API is being worked on so hoping to have it in 2.9.0 but no promises.

**Comment by aquasheep20** [comment]
new tracing does work, Apim-trace-id header will be returned if you set observability.logs.std.level = debug

(I did not test with container < v2.7.1)

|logs.std.level | Ocp-Apim-Debug | apim-trace-id Returned?|
|---:|---:|---:|
|debug | Yes | Yes|
|info | Yes | No|
|warn / error | Yes | No|
|Any | No | No|

**Comment by gentlestfoot** [comment]
> new tracing does work, Apim-trace-id header will be returned if you set observability.logs.std.level = debug
> 
> (I did not test with container < v2.7.1)
> | logs.std.level | Ocp-Apim-Debug | apim-trace-id Returned? |
> | -- | -- | -- |
> | debug | Yes | Yes |
> | info | Yes | No |
> | warn / error | Yes | No |
> | Any | No | No |

@aquasheep20 I was not able to replicate this fix in my environment using version 2.9.3054.0, were there any other steps you took that may have had an impact?

---
> The old tracing API still works so you should not encounter any issues and is supported for SHGW. New Tracing API is being worked on so hoping to have it in 2.9.0 but no promises.

@tomkerkhove I don't see tracing mentioned in the release notes for 2.9.0, do you have an updated estimate for when the new tracing API will be implemented for the SHGW?

Thanks

**Comment by qfyra** [comment]
One and a half years later and this is still not working. And on top of that, the old Ocp-Apim-Trace header is seemingly no longer functional. 

Will this ever be fixed? Or is the ability to test APIM policies no longer an important feature?

Should we take this as an indication that you're deprecating self-hosted APIM gateways? Because that's the impression I'm getting. What other conclusions can one possibly draw here?

**Comment by tomkerkhove** [comment]
> The trace button only works for managed gateway given the portal does not know where your self-hosted gateway is running.

Portal support is not supported, new tracing is planned but old is not supposed to be broken. If it is, please open a support ticket and they can start helping you.

Self-hosted gateway is not deprecated nor do we have any plans for it. We don't provide GitHub support and is best effort so recommend going to Support next time.
